libgalois/include/katana/analytics/k_core/k_core.h

```diff
@@ -0,0 +1,89 @@
+#ifndef KATANA_LIBGALOIS_KATANA_ANALYTICS_KCORE_KCORE_H_
+#define KATANA_LIBGALOIS_KATANA_ANALYTICS_KCORE_KCORE_H_
+
+#include <iostream>
+
+#include <katana/analytics/Plan.h>
+
+#include "katana/AtomicHelpers.h"
+#include "katana/analytics/Utils.h"
+
+// API
+
+namespace katana::analytics {
+
+/// A computational plan to for k-core, specifying the algorithm and any
+/// parameters associated with it.
+class KCorePlan : Plan {
+public:
+  /// Algorithm selectors for KCore
+  enum Algorithm { kSynchronous, kAsynchronous };
+
+  // Don't allow people to directly construct these, so as to have only one
+  // consistent way to configure.
+private:
+  Algorithm algorithm_;
+  uint32_t k_core_number_;
```

Comment:
Should this be in the plan or be a parameter to the call? It feels to me like it should be a parameter since it dramatically affects the output. It's not just an algorithmic choice.

Reply:
Good catch. Not sure why I included it here.